### PR TITLE
feat(tools): use declarative shadow DOM for dev server

### LINKED ISSUE #2008: tools: use declarative shadow dom in dev server
## Description of the requested feature

@wesruv requests that the dev server (`npm start`) be able to load the component's HTML partials immediately, instead of fetching them via `<html-include>`. This would give browser dev tools a leg up, allowing them to access the last-selected-element across page reloads.

Seems like we could have the dev server itself fetch the HTML and append it to a [`<template shadowroot="open">`](https://web.dev/declarative-shadow-dom/) (firefox and safari will require a [polyfill](https://web.dev/declarative-shadow-dom/#polyfill)). 

Alternatively, [HMR](https://modern-web.dev/docs/dev-server/plugins/hmr/) may suit us here.

### Impacted component(s)

- pfe-tools

### PR DESCRIPTION
**Pull request by bennypowers**
### What I Did

Rewrite dev server to use declarative shadow DOM and to be an MPA.

##### Dev Server:
- remove SPA code from demo.ts
- calculate server-side variables in the dev-server
- inject demo markup into a declarative shadow dom template
- inject demo script into the `<head>`
- add declarative shadow DOM polyfill for FF and Safari

### Testing instructions

- `npm start`

Closes #2008

### CODE REVIEW

**Comment by bennypowers** [comment]
draft: still need to figure out how to add files to the filewatcher

**Comment by bennypowers** [comment]
[06534a6](https://github.com/patternfly/patternfly-elements/pull/2017/commits/06534a6bcc62c8b85133d8d5a94db5560c452c67) fixed the file watcher

File: tools/pfe-tools/demo/demo.ts [hunk 1]
```diff
@@ -231,3 +79,7 @@ nav.addEventListener('keydown', event => {
     }
   }
 });
+
+attachShadowRoots(document);
+document.documentElement.toggleAttribute('maximized', localStorage.getItem('pfe-demo-maximized') === 'true');
```
Comment (zeroedin) [inline tools/pfe-tools/demo/demo.ts:84]: Nice little bonus feature with the saved state! 👍

**Review by zeroedin — COMMENTED** [review]
This looks and works great.  Nice work!

**Comment by wesruv** [comment]
I checked out the branch, installed, and ran `npm start` and it is _lightning fast_ which is awesome.

I navigated to pfe-card, and added "wakka" to part="header"
```
      <div class="pfe-card__header ${classMap(classes)}" part="header wakka">
```

But the change didn't show up in the browser, am I running the right command to rebuild on save?

**Comment by bennypowers** [comment]
@wesruv I couldn't reproduce. I suspect you just needed to 
```
npm run clean
npm run build -w @patternfly/pfe-tools
npm start
```

In the interest of time, I'm going to merge now. Any issues that come up please don't hesitate to file them.

**Comment by wesruv** [comment]
Ahh, good to know!

**Pull request merged**